tests/new_user.js

```diff
@@ -0,0 +1,218 @@
+  call_new_user,
+  MINDFUL,
+  KCoin,
+} from "./utils.js";
+import anyTest from "ava";
+import { Keyring } from "@polkadot/api";
+import { mnemonicGenerate } from "@polkadot/util-crypto";
+
+const test = anyTest;
+
+test.beforeEach(async (t) => {
+  // Setup connection to node
+  t.context.api = await init();
+  // Generating random username
+  t.context.username = randomString(5);
+  // Generating random phone number
+  t.context.phoneNumber = randomString(10, "01234567890");
+  // Setup crypto key manager
+  t.context.keyring = new Keyring({ type: "sr25519" });
+  // Generating mnemonic and keys pair
+  const mnemonic = mnemonicGenerate();
+  t.context.pair = t.context.keyring.addFromUri(mnemonic, {
+    name: t.context.name,
+  });
+  // Add Alice private keys
+  t.context.alice = t.context.keyring.addFromUri("//Alice");
+  // Add Bob private keys
+  t.context.bob = t.context.keyring.addFromUri("//Bob");
+
+  // Set tests timeout for 5 minutes
+  t.timeout(300000);
+});
+
+test("Signup", async (t) => {
+  // Call `new_user` tx to register user
+  await call_new_user(
+    t.context.api,
+    t.context.pair,
+    t.context.username,
+    t.context.phoneNumber
+  );
+
+  // Get information about user by `AccountId`
+  const info = await t.context.api.rpc.identity.getUserInfoByAccount.raw(
+    t.context.pair.address
+  );
+
+  t.assert(info.account_id === t.context.pair.address);
+  t.assert(info.user_name === t.context.username);
+  t.assert(info.mobile_number === t.context.phoneNumber);
+});
+
+test("Signup when already signed up on a different device", async (t) => {
+  // Call `new_user` tx to register user
+  await call_new_user(
+    t.context.api,
+    t.context.pair,
+    t.context.username,
+    t.context.phoneNumber
+  );
+
+  // Get information about user by `AccountId`
+  const info = await t.context.api.rpc.identity.getUserInfoByAccount.raw(
+    t.context.pair.address
+  );
+
+  t.assert(info.account_id === t.context.pair.address);
+  t.assert(info.user_name === t.context.username);
+  t.assert(info.mobile_number === t.context.phoneNumber);
+
+  // Generating new mnemonic and keys pair
+  const mnemonic = mnemonicGenerate();
+  t.context.pair = t.context.keyring.addFromUri(mnemonic, {
+    name: t.context.name,
+  });
+
+  // Call `new_user` tx to register user
+  await call_new_user(
+    t.context.api,
+    t.context.pair,
+    t.context.username,
+    t.context.phoneNumber
+  );
+
+  // Get information about user by `AccountId`
+  const infoAfterUpdate =
+    await t.context.api.rpc.identity.getUserInfoByAccount.raw(
+      t.context.pair.address
+    );
+  t.assert(infoAfterUpdate.account_id === t.context.pair.address);
+  t.assert(infoAfterUpdate.user_name === t.context.username);
+  t.assert(infoAfterUpdate.mobile_number === t.context.phoneNumber);
+});
+
+test("Appreciation to a non-user, that person signs up and the appreciation is executed post signup and referral gets the referral reward.", async (t) => {
```

Comment:
I think that this test should work as follows:

1. create account w phone for alice
2. alice sends appreciation to phone number that has no account on chain. verify tx is not executed.
3. bob signs up using phone number.
4. appreciation should be executed and be on chain.
5. both alice and bob balances check - to verify signup reward and referral reward